packages/shell-api/src/database.ts

```diff
@@ -365,11 +365,19 @@ export default class Database extends ShellApiWithMongoClass {
       cmd = { [cmd]: 1 };
     }
 
-    const hiddenCommands = new RegExp(HIDDEN_COMMANDS);
-    if (!Object.keys(cmd).some((k) => hiddenCommands.test(k))) {
-      this._emitDatabaseApiCall('runCommand', { cmd, options });
+    try {
+      const hiddenCommands = new RegExp(HIDDEN_COMMANDS);
+      if (!Object.keys(cmd).some((k) => hiddenCommands.test(k))) {
+        this._emitDatabaseApiCall('runCommand', { cmd, options });
+      }
+      return await this._runCommand(cmd, options);
+    } catch (error: any) {
+      if (error.codeName === 'NotPrimaryNoSecondaryOk') {
+        const message = `not primary - consider passing the readPreference option e.g. db.runCommand({command}, { readPreference: "secondaryPreferred"})`;
```

Comment:
```suggestion
        const message = `not primary - consider passing the readPreference option e.g. db.runCommand({ command }, { readPreference: 'secondaryPreferred' })`;
```

Optional, but I think it's worth being consistent about the spacing around `{` and `}` (the quotes are even less important, but we've moved towards being consistent about single quotes recently e.g. in https://github.com/mongodb-js/mongosh/commit/cbbd2229d3e0e4cf39ea218c5f99e29bc3344581